jenkins/Jenkinsfile

```diff
@@ -0,0 +1,321 @@
+                  curl https://cloudify-release-eu.s3.amazonaws.com/cloudify/${env.VERSION}/${env.PRERELEASE}-release/cloudify-rabbitmq-${env.VERSION}-${env.PRERELEASE}.el7.x86_64.rpm
+                  """, label: 'Fetch RPM built on Jenkins located on S3'
+              }
+            }
+          }
+        }
+        stage('build_prometheus_rpms') {
+          steps {
+            sh script: "mkdir -p ${env.WORKSPACE}/prometheus && cp -rf ${env.WORKSPACE}/${env.PROJECT}/. ${env.WORKSPACE}/prometheus", label: "copying repo to seperate workspace"
+            container('rpmbuild') {
+              dir("${env.WORKSPACE}/prometheus") {
+                echo 'Link items from workspace under rpm sources dir'
+                sh """
+                  rm ~/rpm -fr
+                  ln -s ${env.WORKSPACE}/rpms ./rpms
+                  ln -s . ~/rpm
+                  """
+                echo 'Install rpmdevtools and build dependencies'
+                sh '''
+                  yum install rpmdevtools -y
+                  yum-builddep -y packaging/install_rpm.spec
+                  '''
+                echo 'build rpms'
+                sh '''
+                  rpmbuild -bb packaging/prometheus.spec
+                  rpmbuild -bb packaging/prometheus_node_exporter.spec
+                  rpmbuild -bb packaging/prometheus_blackbox_exporter.spec
+                  rpmbuild -bb packaging/prometheus_postgres_exporter.spec
+                  '''
+                sh script:("mkdir -p ${env.WORKSPACE}/prometheus-rpms && cp -rf /root/rpmbuild/RPMS/x86_64/*.x86_64.rpm ${env.WORKSPACE}/prometheus-rpms"), label:'Copy RPMS to rpms folder'
+              }
+            }
+          }
+          post {
+            success {
+              archiveArtifacts '**/prometheus-rpms/*.rpm'
+            }
+          }
+        }
+      }
+    }
+    stage('build_rpm & deploy ec2 instance') {
+      parallel {
+        stage('build_rpm') {
+          steps {
+            container('rpmbuild') {
+              sh """
+                cd && cd rpmbuild
+                git clone --single-branch --branch ${env.BRANCH_NAME} https://github.com/cloudify-cosmo/cloudify-manager-install.git SOURCES && cd SOURCES
+                """
+              echo 'Setup Github SSH key'
+              setupGithubSSHKey()
+              echo 'Link items from workspace under rpm sources dir'
+              sh """
+                cd /root/rpmbuild/SOURCES
+                rm ~/rpm -fr
+                ln -s ${env.WORKSPACE}/rpms ./rpms
+                cp -a ${env.WORKSPACE}/prometheus-rpms/prometheus-*.rpm ./rpms/
+                cp -a ${env.WORKSPACE}/prometheus-rpms/*_exporter-*.rpm ./rpms/
+                ln -s . ~/rpm
+                """
+              sh script: """
+                pushd rpms
+                  /root/rpmbuild/SOURCES/packaging/fetch_requirements --edition premium -b ${BRANCH_NAME} >~/fetch_requirements.log
```

Comment:
what fetch requirements do? it does not bring all the RPMs? like in previous tage?

Reply:
I'll investigate it

Reply:
It checks if there's an artifact of those RPMs under the same branch in the other builds and displays is